rsp/runtimes/bundles/org.jboss.tools.rsp.server.generic/src/main/java/org/jboss/tools/rsp/server/generic/servertype/actions/EditServerConfigurationActionHandler.java

```diff
@@ -0,0 +1,174 @@
+package org.jboss.tools.rsp.server.generic.servertype.actions;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+
+import org.jboss.tools.rsp.api.DefaultServerAttributes;
+import org.jboss.tools.rsp.api.ServerManagementAPIConstants;
+import org.jboss.tools.rsp.api.dao.ServerActionRequest;
+import org.jboss.tools.rsp.api.dao.ServerActionWorkflow;
+import org.jboss.tools.rsp.api.dao.WorkflowPromptDetails;
+import org.jboss.tools.rsp.api.dao.WorkflowResponse;
+import org.jboss.tools.rsp.api.dao.WorkflowResponseItem;
+import org.jboss.tools.rsp.eclipse.core.runtime.IPath;
+import org.jboss.tools.rsp.eclipse.core.runtime.IStatus;
+import org.jboss.tools.rsp.eclipse.core.runtime.Path;
+import org.jboss.tools.rsp.eclipse.core.runtime.Status;
+import org.jboss.tools.rsp.launching.memento.JSONMemento;
+import org.jboss.tools.rsp.server.generic.impl.Activator;
+import org.jboss.tools.rsp.server.generic.servertype.GenericServerBehavior;
+import org.jboss.tools.rsp.server.model.AbstractServerDelegate;
+import org.jboss.tools.rsp.server.spi.util.StatusConverter;
+
+public class EditServerConfigurationActionHandler {
+	public static final String ACTION_ID = "EditServerConfigurationActionHandler.actionId";
+	public static final String ACTION_LABEL = "Edit Configuration File...";
+	public static final String ACTION_EDIT_FILE_PROMPT_ID = "EditServerConfigurationActionHandler.selection.id";
+	public static final String ACTION_EDIT_FILE_PROMPT_LABEL = "EditServerConfigurationActionHandler.selection.label";
+	
+	public static final ServerActionWorkflow getInitialWorkflow(GenericServerBehavior genericServerDelegate2) {
+		return new EditServerConfigurationActionHandler(genericServerDelegate2).getInitialWorkflowInternal();
+	}
+	
+	private GenericServerBehavior genericServerDelegate;
+	public EditServerConfigurationActionHandler(GenericServerBehavior genericServerDelegate) {
+		this.genericServerDelegate = genericServerDelegate;
+	}
+	
+	protected ServerActionWorkflow getInitialWorkflowInternal() {
+		String[] possiblePaths = getConfigurationFilesRelative();
+		ArrayList<String> asList = new ArrayList<String>();
+		String home = getDefaultWorkingDirectory();
+
+		if( possiblePaths != null ) {
+			for( int i = 0; i < possiblePaths.length; i++ ) {
+				IPath tmpPath = new Path(home).append(possiblePaths[i]);
```

Comment:
idk if the json file will be only edited by us or not but i would add a trim function to prevent some error with spaces. It's common ppl write a list of words like "path1, path2". In this case we're going to append " path2" and having a wrong path.

Reply:
+1